snapcraft/internal/parser.py

```diff
@@ -260,24 +256,25 @@ class Options:
 
 
 def _process_wiki_entry(entry, master_parts_list):
+    # return the number of errors encountered
     try:
         data = yaml.load(entry)
     except ScannerError as e:
         logger.warning(
             'Bad wiki entry, possibly malformed YAML: {}'.format(e))
-        return
+        raise InvalidEntryError('Bad wiki entry: {}'.format(entry))
 
     part_name = data.get('project-part')
     if part_name is not None and part_name in master_parts_list:
         logger.warning("Duplicate part found in wiki: {}".format(
             part_name))
-        return
+        raise InvalidEntryError('duplicate entry: {}'.format(entry))
```

Comment:
Also, the warning and exception messages are redundant. Perhaps it would be better to _only_ raise here, and log in the catcher of the exception instead?